dace/libraries/blas/nodes/axpy.py

```diff
@@ -4,229 +4,156 @@
-                                      vecAdd_tasklet,
-                                      dst_conn='x_con',
-                                      memlet=dace.Memlet.simple(x_in.data, '0'))
-
-        vec_add_state.add_memlet_path(y_in,
-                                      vec_map_entry,
-                                      vecAdd_tasklet,
-                                      dst_conn='y_con',
-                                      memlet=dace.Memlet.simple(y_in.data, '0'))
-
-        vec_add_state.add_memlet_path(vecAdd_tasklet,
-                                      vec_map_exit,
-                                      z_out,
-                                      src_conn='z_con',
-                                      memlet=dace.Memlet.simple(
-                                          z_out.data, '0'))
-
-        return vec_add_sdfg
-
-    @staticmethod
-    def expansion(node, state, sdfg):
-        node.validate(sdfg, state)
-        if node.dtype is None:
-            raise ValueError("Data type must be set to expand " + str(node) +
-                             ".")
-        for e in state.in_edges(node):
-            if e.dst_conn == "_x":
-                buffer_size_x = sdfg.arrays[e.data.data].buffer_size
-            elif e.dst_conn == "_y":
-                buffer_size_y = sdfg.arrays[e.data.data].buffer_size
-        for e in state.out_edges(node):
-            if e.src_conn == "_res":
-                buffer_size_res = sdfg.arrays[e.data.data].buffer_size
-        return ExpandAxpyFPGAStreaming.make_sdfg(node.dtype, int(node.veclen),
-                                                 node.n, node.a, buffer_size_x,
-                                                 buffer_size_y, buffer_size_res)
+    def expansion(node, parent_state: SDFGState, parent_sdfg: SDFG, **kwargs):
+        """
+        :param node: Node to expand.
+        :param parent_state: State that the node is in.
+        :param parent_sdfg: SDFG that the node is in.
+        """
+        return ExpandAxpyVectorized.expansion(
+            node,
+            parent_state,
+            parent_sdfg,
+            schedule=dace.ScheduleType.FPGA_Device,
+            **kwargs)
 
 
 @dace.library.node
 class Axpy(dace.sdfg.nodes.LibraryNode):
-    """Executes a * _x + _y. It implements the BLAS AXPY operation
-        a vector-scalar product with a vector addition.
-
-        Implementations:
-        pure: dace primitive based implementation
-        fpga_stream: FPGA implementation optimized for streaming
+    """
+    Implements the BLAS AXPY operation, which computes a*x + y, where the
+    vectors x and y are of size n. Expects input connectrs "_x" and "_y", and
+    output connector "_res".
     """
 
     # Global properties
     implementations = {
         "pure": ExpandAxpyVectorized,
-        "fpga_stream": ExpandAxpyFPGAStreaming
+        "fpga": ExpandAxpyFpga,
     }
-    default_implementation = 'pure'
+    default_implementation = None
```

Comment:
We have a configuration setting in `.dace.conf` to specify the default BLAS implementation. It can also be specified on the level of the BLAS library programmatically. Explicitly specifying it of course takes priority. The hierarchy chain is implemented in the generic `expansion method`